File: dart/isPalindrome.dart
class Solution {
  bool isPalindrome(int x) {

    String num = x.toString();

    String revers = num.split('').reversed.join('');

    return num == revers;
  }
}

void main() {
  Solution solution = Solution();


  int num1 = 121;
  print('Is $num1 a palindrome? ${solution.isPalindrome(num1)}');

  int num2 = -121;
  print('Is $num2 a palindrome? ${solution.isPalindrome(num2)}');
  int num3 = 7;
  print('Is $num3 a palindrome? ${solution.isPalindrome(num3)}');


  int num4 = 123;
  print('Is $num4 a palindrome? ${solution.isPalindrome(num4)}');
}

File: rust/src/is_palindrome.rs
pub struct Solution;

impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 || (x % 10 == 0 && x != 0) {
            return false;
        }

        let (mut rev, mut org) = (0, x);

        while org > 0 {
            rev = (rev * 10) + org % 10;
            org /= 10;
        }

        rev == x
    }
}
